subsys/bluetooth/host/conn.c

```diff
@@ -1695,8 +1712,22 @@ void bt_conn_set_state(struct bt_conn *conn, bt_conn_state_t state)
 			}
 
 			bt_conn_unref(conn);
+		} else if (old_state == BT_CONN_CONNECT_AUTO) {
```

Comment:
Doesn't have to be done in this PR, but this long list of tests and branches for `old_state` should probably be converted to a switch statement.

Reply:
Made a note to change this outside of this PR.